projects/js-packages/components/components/pricing-slider/types.ts

```diff
@@ -0,0 +1,49 @@
+export type PricingSliderProps = {
+	/**
+	 * The wrapper class name of this PricingSlider component.
+	 */
+	className?: string;
+
+	/**
+	 * The maximum value of the slider.
+	 */
+	maxValue?: number;
+
+	/**
+	 * The minimum value of the slider.
+	 */
+	minValue?: number;
+
+	/**
+	 * The initial value of the slider.
+	 */
+	value?: number;
+
+	/**
+	 * The step value of the slider.
+	 */
+	step?: number;
+
+	/**
+	 * Callback called on every value change.
+	 * The function will be called with two arguments, the first being the new value(s) the second being thumb index.
+	 */
+	onChange?: ( value: number ) => void;
+
+	/**
+	 * Callback called before starting to move a thumb. The callback will only be called if the action will result in a change.
+	 * The function will be called with two arguments, the first being the initial value(s) the second being thumb index.
+	 */
+	onBeforeChange?: ( value: number ) => void;
+
+	/**
+	 * Callback called only after moving a thumb has ended. The callback will only be called if the action resulted in a change.
+	 * The function will be called with two arguments, the first being the result value(s) the second being thumb index.
+	 */
+	onAfterChange?: ( value: number ) => void;
+
+	/**
+	 * Node to render on the slider.
+	 */
+	renderThumb?: React.ReactNode;
```

Comment:
This needs to be updated to resolve a build error:

```suggestion
	renderThumb?: ((props: any, state: any) => Element);
```

Reply:
Pushed up an attempted fix with [332a5ea](https://github.com/Automattic/jetpack/pull/31425/commits/332a5ea8b2074d57ef0e7739c86ee49d2afbda09).